src/Build.UnitTests/BackEnd/SdkResolverLoader_Tests.cs

```diff
@@ -500,5 +563,65 @@ protected override void LoadResolvers(string resolverPath, ElementLocation locat
                 base.LoadResolvers(resolverPath, location, resolvers);
             }
         }
+
+        private sealed class TestableSdkResolverLoader : SdkResolverLoader
+        {
+            public bool MockRunningInVisualStudio { get; set; } = false;
+            public string TestSdkResolversPath { get; set; }
+
+            internal override IReadOnlyList<SdkResolver> LoadAllResolvers(ElementLocation location)
+            {
+                // Store original environment
+                var originalEnvironment = BuildEnvironmentHelper.Instance;
+                
+                try
+                {
+                    // Create a mock BuildEnvironment that returns our controlled value
+                    var mockEnvironment = new BuildEnvironment(
```

Comment:
Does this environment logic need to be part of the resolver? Consider adding it to the test itself.